emc2/simulator/subcolumn.py

```diff
@@ -254,11 +254,10 @@ def set_precip_sub_col_frac(model, is_conv, N_columns=None, use_rad_logic=True,
         precip_type_full = 'convective'
         if use_rad_logic:
             method_str = "Radiation logic"
-            i = 0
             for hyd_type in precip_types:
                 data_frac.append(
                     model.ds[model.conv_frac_names_for_rad[hyd_type]])
-                data_frac[i] = data_frac[i].where(
+                data_frac[-1] = data_frac[i].where(
```

Comment:
Noticed that the flake8 fails now because:
data_frac[-1] = data_frac[i].where(
should become:
data_frac[-1] = data_frac[-1].where(
in this and the next instance.
Also note that there is a redundant i += 1.